src/bin/cobalt/new.rs

```diff
@@ -436,6 +436,37 @@ fn prepend_date_to_filename(
     Ok(())
 }
 
+fn move_from_drafts_to_posts(
+    config: &cobalt_model::Config,
+    file: &path::Path,
+) -> Result<path::PathBuf> {
+    let mut posts = config.posts.clone();
+    posts.include_drafts = true;
+    let posts = posts.build()?;
+    let drafts = posts.drafts.clone();
+
+    if posts
+        .drafts
+        .map(|d| d.includes_file(&file))
+        .unwrap_or_default()
+    {
+        let drafts = drafts.expect("Should have drafts in this if body");
```

Comment:
Can't remember if condition types can be mixed.  Can you do
```rust
if let Some(ref drafts) = posts.drafts && drafts.includes_file(&file)
```

Reply:
```
rror: ambiguous use of `&&`
   --> src\bin\cobalt\new.rs:448:31
    |
448 |     if let Some(ref drafts) = posts.drafts && drafts.includes_file(&file)
    |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider
adding parentheses: `(posts.drafts && drafts.includes_file(&file))`
    |
    = note: this will be a error until the `let_chains` feature is stabilized
    = note: see rust-lang/rust#53668 for more information

error[E0308]: mismatched types
   --> src\bin\cobalt\new.rs:448:31
    |
448 |     if let Some(ref drafts) = posts.drafts && drafts.includes_file(&file)
    |                               ^^^^^^^^^^^^ expected bool, found enum `std::option::Option`
    |
    = note: expected type `bool`
               found type `std::option::Option<cobalt::cobalt_model::files::Files>`

error[E0599]: no method named `includes_file` found for type `std::option::Option<cobalt::cobalt_model::files::Files>` in the current scope
   --> src\bin\cobalt\new.rs:448:54
    |
448 |     if let Some(ref drafts) = posts.drafts && drafts.includes_file(&file)
    |                                                      ^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> src\bin\cobalt\new.rs:448:12
    |
448 |     if let Some(ref drafts) = posts.drafts && drafts.includes_file(&file)
    |            ^^^^^^^^^^^^^^^^ expected bool, found enum `std::option::Option`
    |
    = note: expected type `bool`
               found type `std::option::Option<_>`

```

I tried 
```rust
if (let Some(ref drafts) = posts.drafts) && drafts.includes_file(&file)
```
```
rror: expected expression, found statement (`let`)
   --> src\bin\cobalt\new.rs:448:9
    |
448 |     if (let Some(ref drafts) = posts.drafts) && drafts.includes_file(&file)
    |         ^^^ expected expression
    |
    = note: variable declaration using `let` is a statement
```

Reply:
Yeah, what I expected.  I'm mixed.  I tend to not like this approach but two layers of `if` might also be less than ideal.  Bleh.  I'll go with whatever you prefer here.